com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -289,6 +289,17 @@ public static ObservationProto GetObservationProto(this ISensor sensor, Observat
             }
             else
             {
+                // Check capabilities if we need to concatenate PNGs
+                if (sensor.GetCompressionType() == SensorCompressionType.PNG && shape.Length >= 3 &&
+                    shape[shape.Length - 1] > 3)
```

Comment:
Are we assuming any shape >=3 will work? I am not sure Python will support. I would say shape.Length == 3

Reply:
We can't handle higher dimensional tensors in training, or in barracuda. Theoretically if you had your own trainer and weren't using inference, it could work. Then again, you wouldn't use PNG compression, so I'll simplify this.